install/installer/cluster_dev.py

```diff
@@ -372,6 +497,367 @@ def choose_git_provider(cli_arg, repo):
+    """
+    if not config:
+        cloud_login = ask_user('cloud_login')
+        return cloud_login
+
+    cloud_login = config.get(config_section, 'aws_access_key_id')
+
+    return cloud_login
+
+
+@typechecked
+def get_aws_password(config: {object, bool}, config_section: {str, bool}) -> str:
+    """
+    Get cloud programatic password from settings or from user input
+
+    Args:
+        config (ConfigParser obj|False): INI config
+        config_section (str|False): INI config section
+    Returns:
+        cloud_password string
+    """
+
+    if not config:
+        cloud_password = ask_user('cloud_password')
+        return cloud_password
+
+    cloud_password = config.get(config_section, 'aws_secret_access_key')
+
+    return cloud_password
+
+
+@typechecked
+def get_aws_session(
+        config: {object, bool},
+        config_section: {str, bool},
+        mfa_disabled: str = '',
+) -> str:
+    """
+    Get cloud session from settings or from user input
+
+    Args:
+        config (ConfigParser obj|False): INI config
+        config_section (str|False): INI config section
+        mfa_disabled (str): CLI argument provided by user.
+            If not provided - it set to empty string
+    Returns:
+        session_token string
+    """
+    # If login provided but session - not, user may not have MFA enabled
+    if mfa_disabled:
+        print('SESSION_TOKEN not found, try without MFA')
+        return ''
+
+    if not config:
+        session_token = ask_user('aws_session_token')
+        return session_token
+
+    try:
+        session_token = config.get(config_section, 'aws_session_token')
+    except NoOptionError:
+        print('SESSION_TOKEN not found, try without MFA')
+        return ''
+
+    return session_token
+
+
+@typechecked
+def create_aws_user_and_permissions(user: str, login: str, password: str, session: str) -> dict:
+    """
+    Create cloud user and attach needed permissions
+
+    Args:
+        user (str): Cluster.dev user name
+        login (str): Cloud programatic login
+        password (str): Cloud programatic password
+        session (str): Cloud session token
+    Returns:
+        dict {'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}
+    """
+    iam = boto3.client(
+        'iam',
+        aws_access_key_id=login,
+        aws_secret_access_key=password,
+        aws_session_token=session,
+    )
+    keys_created = False
+    # 'If specified user exist, -clogin and -cpwd will be try use as it
+    # AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+    try:
+        # Required "iam:ListAccessKeys"
+        response = iam.list_access_keys(UserName=user)
+
+        if response['AccessKeyMetadata'][0]['AccessKeyId'] == login:
+            print(f'Specified creds belong to exist user {user}, use it')
+
+            return {
+                'key': login,
+                'secret': password,
+                'created': keys_created,
+            }
```

Comment:
Move this out or relocate to the else-block. It's a bad idea to expect an exception on multiple lines because it'll make your app more fragile shadowing any future instances of this exception appearing on other lines.